src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer.cs

```diff
@@ -139,6 +162,16 @@ public sealed class DiagnosticDescriptorCreationAnalyzer : DiagnosticAnalyzer
             description: s_localizableProvideCustomTagsDescription,
             customTags: WellKnownDiagnosticTags.Telemetry);
 
+        public static readonly DiagnosticDescriptor DoNotUseReservedDiagnosticIdRule = new DiagnosticDescriptor(
+            DiagnosticIds.DoNotUseReservedDiagnosticIdRuleId,
+            s_localizableDoNotUseReservedDiagnosticIdTitle,
+            s_localizableDoNotUseReservedDiagnosticIdMessage,
+            DiagnosticCategory.MicrosoftCodeAnalysisDesign,
+            DiagnosticSeverity.Error,
```

Comment:
❗️ Diagnostic analyzers cannot have default severity error. It breaks the analyzer driver's suppression feature. Users who want to see analyzer diagnostics as errors must configure the severity through .editorconfig, a rule set file, and/or set warnings as errors during the build.

Reply:
> No diagnostic analyzers can have default severity error

I don't believe that is true here. We even talked about having this diagnostic in the compiler itself, and settled as having this an analyzer diagnostic with severity error. Tagging @terrajobst

Reply:
➡️ Filed #3159 to fix this

Reply:
@mavasani The limitation here is not specific to the analyzer in question. Setting a default severity of Error breaks a core compiler feature designed to support scenarios where a team finds an analyzer false positive.

Reply:
I don't understand - analyzer errors can be suppressed just like analyzer warning. Default severity error just ensures a higher visibility for analyzer authors using reserved diagnostic IDs. Additionally, I think we want analyzer authors to absolutely stop using reserved diagnostic IDs